src/mbgl/style/style.cpp

```diff
@@ -166,6 +173,8 @@ void Style::cascade() {
     for (const auto& layer : layers) {
         layer->cascade(parameters);
     }
+
+    loadSources();
```

Comment:
I think `loadSources()` is not necessary here -- `cascade` doesn't change `source->enabled` so it can never trigger a source being needed to load.